diff --git src/dawn/common/Platform.h src/dawn/common/Platform.h
index dc8d79c01b..1623fa229b 100644
--- src/dawn/common/Platform.h
+++ src/dawn/common/Platform.h
@@ -117,10 +117,10 @@
 
 #elif defined(__loongarch__)
 #define DAWN_PLATFORM_IS_LOONGARCH 1
-#if _LOONGARCH_SIM == _ABILP32
-#define DAWN_PLATFORM_IS_LOONGARCH32 1
-#else
+#if defined(__loongarch64)
 #define DAWN_PLATFORM_IS_LOONGARCH64 1
+#else
+#define DAWN_PLATFORM_IS_LOONGARCH32 1
 #endif
 
 #elif defined(__riscv)
