diff --git include/openssl/base.h include/openssl/base.h
index 80d18a78d..724f993fd 100644
--- include/openssl/base.h
+++ include/openssl/base.h
@@ -116,6 +116,9 @@ extern "C" {
 #define OPENSSL_32_BIT
 #elif defined(__myriad2__)
 #define OPENSSL_32_BIT
+#elif defined(__loongarch64)
+#define OPENSSL_64_BIT
+#define OPENSSL_LOONG64
 #else
 // Note BoringSSL only supports standard 32-bit and 64-bit two's-complement,
 // little-endian architectures. Functions will not produce the correct answer
