diff --git include/v8config.h include/v8config.h
index 85bbfce01d..77b837975f 100644
--- include/v8config.h
+++ include/v8config.h
@@ -721,6 +721,8 @@ V8 shared library set USING_V8_SHARED.
 #elif __riscv_xlen == 32
 #define V8_TARGET_ARCH_RISCV32 1
 #endif
+#elif defined(__loongarch64)
+#define V8_TARGET_ARCH_LOONG64 1
 #else
 #error Target architecture was not detected as supported by v8
 #endif
