diff --git BUILD.gn BUILD.gn
index 42fac07c63..70cee80d58 100644
--- BUILD.gn
+++ BUILD.gn
@@ -313,6 +313,10 @@ target(link_target_type, "ffmpeg_internal") {
         asmflags += [ "-DHAVE_VFP_ARGS=0" ]
       }
     }
+
+    if (current_cpu == "loong64") {
+      cflags += [ "-mlasx" ]
+    }
   }
 
   if (is_fuchsia || (is_posix && !is_mac)) {
diff --git ffmpeg_generated.gni ffmpeg_generated.gni
index 30aa23c288..f941cfff08 100644
--- ffmpeg_generated.gni
+++ ffmpeg_generated.gni
@@ -462,6 +462,12 @@ if ((is_android && current_cpu == "arm64") || (is_apple && current_cpu == "arm64
   ]
 }
 
+if (use_linux_config && current_cpu == "loong64") {
+  ffmpeg_c_sources += [
+    "libavutil/loongarch/cpu.c",
+  ]
+}
+
 if ((is_android && current_cpu == "arm" && arm_use_neon) || (use_linux_config && current_cpu == "arm" && arm_use_neon) || (use_linux_config && current_cpu == "arm")) {
   ffmpeg_c_sources += [
     "libavcodec/arm/flacdsp_init_arm.c",
@@ -493,6 +499,21 @@ if ((is_apple && current_cpu == "arm64") || (is_win && current_cpu == "arm64") |
   ]
 }
 
+if (use_linux_config && current_cpu == "loong64") {
+  ffmpeg_c_sources += [
+    "libavcodec/loongarch/h264_intrapred_init_loongarch.c",
+    "libavcodec/loongarch/hpeldsp_init_loongarch.c",
+    "libavcodec/loongarch/hpeldsp_lasx.c",
+    "libavcodec/loongarch/videodsp_init.c",
+    "libavcodec/loongarch/vp8dsp_init_loongarch.c",
+    "libavcodec/loongarch/vp8_lpf_lsx.c",
+    "libavcodec/loongarch/vp8_mc_lsx.c",
+  ]
+  ffmpeg_gas_sources += [
+    "libavcodec/loongarch/h264intrapred.S",
+  ]
+}
+
 if ((is_apple && current_cpu == "arm64" && ffmpeg_branding == "Chrome") || (is_win && current_cpu == "arm64" && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "arm64" && ffmpeg_branding == "Chrome") || (use_linux_config && current_cpu == "arm64" && ffmpeg_branding == "ChromeOS")) {
   ffmpeg_c_sources += [
     "libavcodec/aarch64/h264chroma_init_aarch64.c",
@@ -507,6 +528,25 @@ if ((is_apple && current_cpu == "arm64" && ffmpeg_branding == "Chrome") || (is_w
   ]
 }
 
+if (use_linux_config && current_cpu == "loong64" && ffmpeg_branding == "Chrome") {
+  ffmpeg_c_sources += [
+    "libavcodec/loongarch/h264chroma_init_loongarch.c",
+    "libavcodec/loongarch/h264dsp_init_loongarch.c",
+    "libavcodec/loongarch/h264_deblock_lasx.c",
+    "libavcodec/loongarch/h264idct_loongarch.c",
+    "libavcodec/loongarch/h264dsp_lasx.c",
+    "libavcodec/loongarch/h264qpel_init_loongarch.c",
+    "libavcodec/loongarch/h264qpel_lsx.c",
+    "libavcodec/loongarch/h264qpel_lasx.c",
+  ]
+  ffmpeg_gas_sources += [
+    "libavcodec/loongarch/h264chroma_asm.S",
+    "libavcodec/loongarch/h264dsp_asm.S",
+    "libavcodec/loongarch/h264idct_asm.S",
+    "libavcodec/loongarch/h264qpel_asm.S",
+  ]
+}
+
 if (use_linux_config && current_cpu == "arm" && arm_use_neon && ffmpeg_branding == "ChromeOS") {
   ffmpeg_c_sources += [
     "libavcodec/arm/blockdsp_init_neon.c",
